boot/boottest/mockbootloader.go

```diff
@@ -70,3 +72,21 @@ func (b *MockBootloader) Name() string {
 func (b *MockBootloader) ConfigFile() string {
 	return filepath.Join(b.bootdir, "mockboot/mockboot.cfg")
 }
+
+func (b *MockBootloader) ExtractKernelAssets(s *snap.Info, snapf snap.Container) error {
+	// XXX: should we use "kernel.yaml" for this?
+	var forceKernelExtraction bool
+	if _, err := snapf.ReadFile("meta/force-kernel-extraction"); err == nil {
```

Comment:
This function here is problematic - with that the `kernel_os_test.go` tests really just this mock, we have no idea if the real "grub" or "uboot" implementations will work. It will also easily get out of sync.

I think we need to call the real "grub/uboot/..." implementations here.

Reply:
@mvo5 please check updated tests. They are now using real glue/uboot bootloader to test bootloader functionality. Mocbootloader is only used for tests on classic system